### ISSUE TITLE
There were some errors while trying to build your project

### ISSUE THREAD

**Issue by martinkearn** [manifoldjs/ManifoldJS#86]
I seem to be getting this error ("There were some errors while trying to build your project") when trying to build a manifest. 

I've tried multiple browsers and urls.

I'm just doing the bare minimum here - not adding any icons or anything, just the mandatory fields.

I'm guessing there are mandatory fields that are required but not enforced in the forms?

**Comment by msrodri** [comment]
Thanks for reporting this. We'll take a look at the issue.

**Comment by martinkearn** [comment]
Thanks, I've also now noticed that I'm getting this error in the JavaScript console when trying to generate a manifest:

"HTTP500: SERVER ERROR - The server encountered an unexpected condition that prevented it from fulfilling the request.
(XHR): POST - http://manifold-api-prod.azurewebsites.net/manifests/"

**Comment by boyofgreen** [comment]
Hey Martin,
  can you send me a URL or two that you were trying?  We did just upgrade the site, so I'm not sure if we broke something or not, I just tried facebook.com and it worked fine.

**Comment by hellokingdom** [comment]
Hi

I've tried to build via manifoldjs.com and via the nodejs command line and I see this error for ios platform:

Installing "cordova-plugin-whitelist" for ios

npm http GET https://registry.npmjs.org/cordova-plugin-whitelist
npm http 304 https://registry.npmjs.org/cordova-plugin-whitelist
Failed to install 'cordova-plugin-whitelist':CordovaError: Plugin doesn't suppor
t this project's cordova-ios version. cordova-ios: 3.8.0, failed version require
ment: >=4.0.0-dev
    at checkEngines (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifo
ldjs\node_modules\cordova\node_modules\cordova-lib\src\plugman\install.js:119:29
)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\cordova-lib\src\plugman\install.js:296:16
    at _fulfilled (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifold
js\node_modules\cordova\node_modules\q\q.js:787:54)
    at self.promiseDispatch.done (C:\Users\jameshyatt\AppData\Roaming\npm\node_m
odules\manifoldjs\node_modules\cordova\node_modules\q\q.js:816:30)
    at Promise.promise.promiseDispatch (C:\Users\jameshyatt\AppData\Roaming\npm\
node_modules\manifoldjs\node_modules\cordova\node_modules\q\q.js:749:13)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\q\q.js:557:44
    at flush (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\no
de_modules\cordova\node_modules\q\q.js:108:17)
    at process._tickCallback (node.js:419:13)
Failed to install 'cordova-plugin-whitelist':CordovaError: Plugin doesn't suppor
t this project's cordova-ios version. cordova-ios: 3.8.0, failed version require
ment: >=4.0.0-dev
    at checkEngines (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifo
ldjs\node_modules\cordova\node_modules\cordova-lib\src\plugman\install.js:119:29
)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\cordova-lib\src\plugman\install.js:296:16
    at _fulfilled (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifold
js\node_modules\cordova\node_modules\q\q.js:787:54)
    at self.promiseDispatch.done (C:\Users\jameshyatt\AppData\Roaming\npm\node_m
odules\manifoldjs\node_modules\cordova\node_modules\q\q.js:816:30)
    at Promise.promise.promiseDispatch (C:\Users\jameshyatt\AppData\Roaming\npm\
node_modules\manifoldjs\node_modules\cordova\node_modules\q\q.js:749:13)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\q\q.js:557:44
    at flush (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\no
de_modules\cordova\node_modules\q\q.js:108:17)
    at process._tickCallback (node.js:419:13)

WARNING: Failed to add the Cordova platforms: ios.
Command failed: npm http GET https://registry.npmjs.org/cordova-plugin-whitelist

npm http 304 https://registry.npmjs.org/cordova-plugin-whitelist
Failed to install 'cordova-plugin-whitelist':CordovaError: Plugin doesn't suppor
t this project's cordova-ios version. cordova-ios: 3.8.0, failed version require
ment: >=4.0.0-dev
    at checkEngines (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifo
ldjs\node_modules\cordova\node_modules\cordova-lib\src\plugman\install.js:119:29
)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\cordova-lib\src\plugman\install.js:296:16
    at _fulfilled (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifold
js\node_modules\cordova\node_modules\q\q.js:787:54)
    at self.promiseDispatch.done (C:\Users\jameshyatt\AppData\Roaming\npm\node_m
odules\manifoldjs\node_modules\cordova\node_modules\q\q.js:816:30)
    at Promise.promise.promiseDispatch (C:\Users\jameshyatt\AppData\Roaming\npm\
node_modules\manifoldjs\node_modules\cordova\node_modules\q\q.js:749:13)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\q\q.js:557:44
    at flush (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\no
de_modules\cordova\node_modules\q\q.js:108:17)
    at process._tickCallback (node.js:419:13)
Failed to install 'cordova-plugin-whitelist':CordovaError: Plugin doesn't suppor
t this project's cordova-ios version. cordova-ios: 3.8.0, failed version require
ment: >=4.0.0-dev
    at checkEngines (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifo
ldjs\node_modules\cordova\node_modules\cordova-lib\src\plugman\install.js:119:29
)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\cordova-lib\src\plugman\install.js:296:16
    at _fulfilled (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifold
js\node_modules\cordova\node_modules\q\q.js:787:54)
    at self.promiseDispatch.done (C:\Users\jameshyatt\AppData\Roaming\npm\node_m
odules\manifoldjs\node_modules\cordova\node_modules\q\q.js:816:30)
    at Promise.promise.promiseDispatch (C:\Users\jameshyatt\AppData\Roaming\npm\
node_modules\manifoldjs\node_modules\cordova\node_modules\q\q.js:749:13)
    at C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\node_modu
les\cordova\node_modules\q\q.js:557:44
    at flush (C:\Users\jameshyatt\AppData\Roaming\npm\node_modules\manifoldjs\no
de_modules\cordova\node_modules\q\q.js:108:17)
    at process._tickCallback (node.js:419:13)

One or more errors occurred when generating the application.

**Comment by boyofgreen** [comment]
The website is clearly failing due to the iOS issue as well.  I am using this test site: https://sentimentalweb.azurewebsites.net/ and breaks every time on the website, errors on iOS in CLI, runs fine without iOS on CLI. 
We found that a change was pulled out of the cordova release at the last moment and that is breaking our builds.  We're working on 0.3.1 that will put a temporary fix in for this until the new cordova release.  Hold tight for release details.

**Comment by f2bo** [comment]
This was caused by a recent update in Cordova’s whitelist plugin that requires a version of cordova-ios that is not released yet. The issue is benign, since the plugin fails to install for the iOS platform, it will continue to use the “native” whitelisting mechanism. However, it breaks any  automated plugin installations, including ManifoldJS.

To work around this issue, we updated ManifoldJS to force the installation of the previous version of the plugin, which allows the generation process to complete without errors. This is a temporary fix, until the issue is rectified by Cordova. Note that this requires users to update ManifoldJS to obtain the fix when we publish it (should be in v0.3.1). 

Users of the ManifoldJS website should also wait until the site is updated to the latest release.

See https://github.com/manifoldjs/ManifoldJS/commit/c4e9e309116a06af28046d18356c205e1c8f91b9.
